docs/spec/identities.md

```diff
@@ -0,0 +1,438 @@
+
+> **TODO** Define how to delegate trust to a set of keys `K \ D` for the purpose
+> of recovering from key revocations in case the quorum rules can no longer be
+> met with the remaining keys. (ie. "Social Recovery")
+
+## Effect on Replication
+
+Peers MUST NOT replicate repositories whose identities they are unable to
+verify. If an update to the identity cannot be verified, they SHOULD keep the
+data they already have, but refuse to update it. Peers SHOULD keep the identity
+histories regardless of their verification status.
+
+When **cloning** a repository, the attestation history according to the remote
+peer is fetched before all other repository contents, and the verification
+procedure is run on it. If this does not yield a verified status, the clone is
+aborted. The resulting repository state MUST include the attestation histories
+of at least a quorum of the delegates as per the remote peer's view of the
+identity document. In `git`, the claim that this will be the case can be
+determined before fetching the repository contents by examining the advertised
+remote refs. If these preconditions are not met, the clone is aborted, and
+already fetched data is pruned.
+
+When **fetching** (i.e. updating) a repository, a predetermined set of verified
+delegates is already known -- their views of the attestation history SHALL be
+fetched first, along with any advertised histories not yet present in the local
+repository (as those might be owned by future delegates). The fetching end MUST
+now find a valid `Doc` revision near the tips of the available delegate
+histories, and pick the most recent one. If two valid identities are found of
+which neither is in the ancestry path of the other, the attestation chain is
+said to have **forked**. In lieu of a consensus system, it is undecidable which
+side of the fork to commit to, and so replication of the attested repository
+SHALL refuse any further updates.
+
+> **TODO** Fork detection (ie. pairwise compare if in same ancestry path, either
+> select most recent or flag as forked)
+
+## Git Encoding
+
+In the `git` implementation, a `Doc` corresponds to a `blob` object, stored as
+the single entry of a `tree` object, such that its name (acc. to the `tree`) is
+equal to the `blob` hash of the _initial_ version of the `Doc`, serialised in
+canonical form. That is:
+
+    let name = git hash-object -t blob doc.canonical_form()
+
+An `Identity` corresponds to a `commit` object.
+
+We map the fields as follows:
+
+```rust
+/* Simplified git object model */
+struct Commit {
+    id: Oid,
+    tree: Tree,
+    message: String,
+}
+
+struct Tree {
+    id: Oid,
+    entries: Vec<TreeEntry>,
+}
+
+struct TreeEntry {
+    id: Oid,
+    name: String,
+    object: Blob, // faithfully
+}
+
+struct Blob {
+    id: Oid,
+    content: Vec<u8>,
+}
+
+/* Mapping (trivial type conversions elided) */
+let commit = /* .. */;
+let identity = Identity {
+    id: commit.id,
+    root: commit.tree.entries[0].name,
```

Comment:
The name of the _initial_ blob is it's own hash (`Oid`). This way, we kind of force everyone to actually run the verification, instead of trusting a `root` field somewhere.

Reply:
Sorry, what I'm saying is that `commit.tree.entries[0].name` points to a `String` not an `Revision`/`Oid`, which seems like a mistake to me.

Is what you're saying the `name` points to the `Blob` and we want to use _its_ `Oid`?

Reply:
I wrote "trivial type conversions elided" because I assumed that the previous definition of `let name` clarifies that the name is actually an `Oid` (the hash of the initial version of the blob). Perhaps at this point I should reconsider the choice of using rust syntax -- specifying this more precisely would get very verbose.

Reply:
I was able to get to the point that it was the blob's initial `Oid`, but thought it might have been a clerical error. Should be fine though :+1: